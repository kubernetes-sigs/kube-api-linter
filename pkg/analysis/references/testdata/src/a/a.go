package a

// TestWithPolicyAllowRefAndRefs tests the linter with PolicyAllowRefAndRefs
// In this mode, Reference/References are flagged ANYWHERE in field names, but Ref/Refs are allowed
type TestWithPolicyAllowRefAndRefs struct {
	// Fields ending with Reference should be flagged
	NodeReference string `json:"nodeReference"` // want `naming convention "reference-to-ref": field NodeReference: field names should use 'Ref' instead of 'Reference'`

	ConfigReference string `json:"configReference"` // want `naming convention "reference-to-ref": field ConfigReference: field names should use 'Ref' instead of 'Reference'`

	// Fields ending with References should be flagged
	NodeReferences []string `json:"nodeReferences"` // want `naming convention "reference-to-ref": field NodeReferences: field names should use 'Ref' instead of 'Reference'`

	ConfigReferences []string `json:"configReferences"` // want `naming convention "reference-to-ref": field ConfigReferences: field names should use 'Ref' instead of 'Reference'`

	// Fields with Reference at beginning should be flagged
	ReferenceCount int `json:"referenceCount"` // want `naming convention "reference-to-ref": field ReferenceCount: field names should use 'Ref' instead of 'Reference'`

	ReferenceData string `json:"referenceData"` // want `naming convention "reference-to-ref": field ReferenceData: field names should use 'Ref' instead of 'Reference'`

	// Fields with References at beginning should be flagged
	ReferencesCount int `json:"referencesCount"` // want `naming convention "reference-to-ref": field ReferencesCount: field names should use 'Ref' instead of 'Reference'`

	ReferencesData []string `json:"referencesData"` // want `naming convention "reference-to-ref": field ReferencesData: field names should use 'Ref' instead of 'Reference'`

	// Fields with Reference in middle should be flagged
	CrossReferenceID string `json:"crossReferenceID"` // want `naming convention "reference-to-ref": field CrossReferenceID: field names should use 'Ref' instead of 'Reference'`

	// Fields with References in middle should be flagged
	CrossReferencesMap map[string]string `json:"crossReferencesMap"` // want `naming convention "reference-to-ref": field CrossReferencesMap: field names should use 'Ref' instead of 'Reference'`

	// Past tense "Referenced" should be flagged (has "Reference")
	Referenced bool `json:"referenced"` // want `naming convention "reference-to-ref": field Referenced: field names should use 'Ref' instead of 'Reference'`

	// Fields with Ref/Refs anywhere are ALLOWED in this mode (no diagnostics expected)
	NodeRef string `json:"nodeRef"`

	ConfigRef string `json:"configRef"`

	NodeRefs []string `json:"nodeRefs"`

	ConfigRefs []string `json:"configRefs"`

	RefCount int `json:"refCount"`

	RefsData []string `json:"refsData"`

	InternalRefData string `json:"internalRefData"`

	InternalRefsData []string `json:"internalRefsData"`

	// Normal fields should not be flagged (no Reference/References/Ref/Refs)
	Name string `json:"name"`

	Namespace string `json:"namespace"`

	// Edge cases - Preference contains "reference" and WILL be flagged with (?i)
	// This is intentional to catch all variations including malformed casing
	PreferenceType string `json:"preferenceType"` // want `naming convention "reference-to-ref": field PreferenceType: field names should use 'Ref' instead of 'Reference'`

	Preferences map[string]string `json:"preferences,omitempty"` // want `naming convention "reference-to-ref": field Preferences: field names should use 'Ref' instead of 'Reference'`

	// These don't contain capital Reference
	Referral string `json:"referral"`

	Referee string `json:"referee"`
}
