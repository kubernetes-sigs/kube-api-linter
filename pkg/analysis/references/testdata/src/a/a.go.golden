package a

// TestWithPolicyAllowRefAndRefs tests the linter with PolicyAllowRefAndRefs
// In this mode, Reference/References are flagged ANYWHERE in field names, but Ref/Refs are allowed
type TestWithPolicyAllowRefAndRefs struct {
	// Fields ending with Reference should be flagged
	NodeRef string `json:"nodeRef"` // want `naming convention "reference-to-ref": field NodeReference: field names should use 'Ref' instead of 'Reference'`

	ConfigRef string `json:"configRef"` // want `naming convention "reference-to-ref": field ConfigReference: field names should use 'Ref' instead of 'Reference'`

	// Fields ending with References should be flagged
	NodeRefs []string `json:"nodeRefs"` // want `naming convention "reference-to-ref": field NodeReferences: field names should use 'Ref' instead of 'Reference'`

	ConfigRefs []string `json:"configRefs"` // want `naming convention "reference-to-ref": field ConfigReferences: field names should use 'Ref' instead of 'Reference'`

	// Fields with Reference at beginning should be flagged
	RefCount int `json:"refCount"` // want `naming convention "reference-to-ref": field ReferenceCount: field names should use 'Ref' instead of 'Reference'`

	RefData string `json:"refData"` // want `naming convention "reference-to-ref": field ReferenceData: field names should use 'Ref' instead of 'Reference'`

	// Fields with References at beginning should be flagged
	RefsCount int `json:"refsCount"` // want `naming convention "reference-to-ref": field ReferencesCount: field names should use 'Ref' instead of 'Reference'`

	RefsData []string `json:"refsData"` // want `naming convention "reference-to-ref": field ReferencesData: field names should use 'Ref' instead of 'Reference'`

	// Fields with Reference in middle should be flagged
	CrossRefID string `json:"crossRefID"` // want `naming convention "reference-to-ref": field CrossReferenceID: field names should use 'Ref' instead of 'Reference'`

	// Fields with References in middle should be flagged
	CrossRefsMap map[string]string `json:"crossRefsMap"` // want `naming convention "reference-to-ref": field CrossReferencesMap: field names should use 'Ref' instead of 'Reference'`

	// Past tense "Referenced" should be flagged (has "Reference")
	Refd bool `json:"refd"` // want `naming convention "reference-to-ref": field Referenced: field names should use 'Ref' instead of 'Reference'`

	// Fields with Ref/Refs anywhere are ALLOWED in this mode (no diagnostics expected)
	NodeRef string `json:"nodeRef"`

	ConfigRef string `json:"configRef"`

	NodeRefs []string `json:"nodeRefs"`

	ConfigRefs []string `json:"configRefs"`

	RefCount int `json:"refCount"`

	RefsData []string `json:"refsData"`

	InternalRefData string `json:"internalRefData"`

	InternalRefsData []string `json:"internalRefsData"`

	// Normal fields should not be flagged (no Reference/References/Ref/Refs)
	Name string `json:"name"`

	Namespace string `json:"namespace"`

	// Edge cases - Preference contains "reference" and WILL be flagged with (?i)
	// This is intentional to catch all variations including malformed casing
	PRefType string `json:"pRefType"` // want `naming convention "reference-to-ref": field PreferenceType: field names should use 'Ref' instead of 'Reference'`

	PRefs map[string]string `json:"pRefs,omitempty"` // want `naming convention "reference-to-ref": field Preferences: field names should use 'Ref' instead of 'Reference'`

	// These don't contain capital Reference
	Referral string `json:"referral"`

	Referee string `json:"referee"`
}


